File: BonusServer/Models/CollectData.cs
using FunLobbyUtils;

namespace BonusServer.Models
{
    public class CollectData
    {
        public float TotalBet { get; set; }
        public float TotalWin { get; set; }
        public float WinA { get; set; }
        public float WinB { get; set; }
        public int BonusType { get; set; }

        public CollectData(
            float totalBet = 0,
            float totalWin = 0,
            float winA = 0,
            float winB = 0,
            int bonusType = 0)
        {
            this.TotalBet = totalBet;
            this.TotalWin = totalWin;
            this.WinA = winA;
            this.WinB = winB;
            this.BonusType = bonusType;
        }

        public CollectData Clone()
        {
            CollectData newData = new CollectData(
                this.TotalBet,
                this.TotalWin,
                this.WinA,
                this.WinB,
                this.BonusType
            );
            return newData;
        }

        public void Reset()
        {
            this.TotalBet = 0;
            this.TotalWin = 0;
            this.WinA = 0;
            this.WinB = 0;
        }
    }
}

File: BonusServer_Node/models/collectData.js

export default class CollectData
{
    TotalBet;
    TotalWin;
    WinA;
    WinB;
    BonusType;

    constructor(
        totalBet,
        totalWin,
        winA,
        winB,
        bonusType)
    {
        this.TotalBet = totalBet ? parseInt(totalBet) : 0;
        this.TotalWin = totalWin ? parseInt(totalWin) : 0;
        this.WinA = winA ? parseInt(winA) : 0;
        this.WinB = winB ? parseInt(winB) : 0;
        this.BonusType = bonusType ? parseInt(bonusType) : 0;
    }

    Clone()
    {
        const newData = new CollectData(
            this.TotalBet,
            this.TotalWin,
            this.WinA,
            this.WinB,
            this.BonusType
        );
        return newData;
    }

    Reset()
    {
        this.TotalBet = 0;
        this.TotalWin = 0;
        this.WinA = 0;
        this.WinB = 0;
    }
}